article_store/articles/api.py

```diff
@@ -0,0 +1,206 @@
+from lxml.etree import (
+    Element,
+    SubElement,
+    tostring,
+)
+from django.db.models import QuerySet
+from django.http.response import HttpResponse
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from rest_framework import status, viewsets
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework_xml.renderers import XMLRenderer
+
+from articles.xml_parser import ArticleXMLParser
+
+from articles.models import (
+    Article,
+    ArticleVersion,
+    Content,
+)
+from articles.serializers import (
+    ArticleSerializer,
+    ArticleVersionSerializer,
+)
+
+from articles.utils import parse_accept_language_header
+
+
+class ArticleViewSet(viewsets.ModelViewSet):
+    model = Article
+    queryset = Article.objects.all()
+    serializer_class = ArticleSerializer
+
+
+def article_list_xml_generator(articles: 'QuerySet[Article]') -> Element:
+    """Generate an xml representation of all `Article` objects.
+
+    :param articles: QuerySet[Article]
+    :return: class: `Element`
+    """
+    name_space = {'libero': 'http://libero.pub'}
+
+    root = Element('{%s}articles' % name_space['libero'], nsmap=name_space)
+
+    for article in articles:
+        child = SubElement(root, '{%s}article' % name_space['libero'])
+        child.text = str(article.id)
+
+    return root
+
+
+class ArticleXMLAPIView(APIView):
```

Comment:
this is called `ArticleXML...` but comparing it to `ArticleList...` (which also uses XML) its difference is being oriented to a single article and its components rather than a list. Can be called `ArticleItem` as opposed to `List`, or `ArticleAggregate...` for the DDD-style?
(`...` stands for `APIView`, which is the superclass)

Reply:
So... `ArticleXMLAPIView` -> `ArticleAggregateAPIView` ?